gobblin-metastore/src/main/resources/db/migration/V1_0_1__Add_schema_modification_sprocs.sql

```diff
@@ -0,0 +1,113 @@
+-- Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+-- this file except in compliance with the License. You may obtain a copy of the
+-- License at  http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software distributed
+-- under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+-- CONDITIONS OF ANY KIND, either express or implied.
+
+--
+-- Version: 1.0.1
+--
+-- Helper stored procedures that enable idempotent schema modifications in MySql.
+-- The following stored procedures are being added:
+--   1. add_index_if_not_exists: check if the specified index exists on the
+--                                  given table and create it if it does not.
+--   2. drop_index_if_exists: check if the specified index exists on the
+--                            given table and delete it if it does.
+--   3. add_column_if_not_exists: check if the specified columns exists on the
+--                                give table and create it if it does not.
+--   4. modify_column_if_exists: check if the specified column exists on the
+--                               given table and modifies it if it does.
+--
+
+DELIMITER //
```

Comment:
I'm not really sure how this could be done in all cases.  For instance, mysql doesn't support IF statements outside of sprocs.  Therefore we have to start doing something like the following for every schema change:
```sql
set @col_exists = 0;
SELECT 1 FROM INFORMATION_SCHEMA.COLUMNS
WHERE TABLE_NAME='myTable'
AND column_name='newColumn'
and table_schema = database()
into @col_exists;

set @stmt = case @col_exists
when 0 then CONCAT(
'alter table myTable'
, ' ADD COLUMN newColumn INT NULL DEFAULT 0 AFTER oldColumn'
,';')
else 'select ''column already exists, no op'''
end;

PREPARE stmt FROM @stmt;
EXECUTE stmt;
DEALLOCATE PREPARE stmt;
```

This will result in huge migrations and complex migration scripts.  Is there another way?